File: specification/specs/x_pack/security/role/put_role/PutRoleRequest.ts
@rest_spec_name("security.put_role")
class PutRoleRequest extends RequestBase {
  path_parts?: {
    name: Name;
  }
  query_parameters?: {
    refresh?: Refresh;
  }
  body?: {
    applications?: ApplicationPrivileges[];
    cluster?: string[];
    global?: Dictionary<string, UserDefinedValue>;
    indices?: IndicesPrivileges[];
    metadata?: Dictionary<string, UserDefinedValue>;
    run_as?: string[];
  }
}

File: output/csharp/XPack/Security/Role/PutRole/PutRoleRequest.cs
using Nest.Common;
using Nest.XPack;
using Nest.CommonAbstractions;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.XPack {

	public class PutRoleRequest : RequestBase {
		
		[DataMember(Name="refresh")]
		public Refresh Refresh { get; set; }


		[DataMember(Name="applications")]
		public List<ApplicationPrivileges> Applications { get; set; }


		[DataMember(Name="cluster")]
		public List<string> Cluster { get; set; }


		[DataMember(Name="global")]
		public IDictionary<string, object> Global { get; set; }


		[DataMember(Name="indices")]
		public List<IndicesPrivileges> Indices { get; set; }


		[DataMember(Name="metadata")]
		public IDictionary<string, object> Metadata { get; set; }


		[DataMember(Name="run_as")]
		public List<string> RunAs { get; set; }

	}
}
